### PR TITLE
Add support to build boost with Win64 MSVS generators.

### PR DESCRIPTION
**Pull request by o01eg**
This fixes boost compilation with Win64 MS VS.

This allows to successfully build SDK archive.

### CODE REVIEW

**Comment by adrianbroher** [comment]
Looks good to me from the technical point, but why should there be a 64-bit SDK?

File: CMakeLists.txt [hunk 1]
```diff
@@ -549,8 +549,12 @@ else()
     set(_VS_PLATFORM_TOOLSET_VERSION "${_VS_PLATFORM_TOOLSET_VERSION}.0")
     set(BOOST_CONFIGURE_FLAGS
         debug-store=database
-        address-model=32
     )
+    if(CMAKE_CL_64)
```
Comment (adrianbroher) [inline CMakeLists.txt]: Discouraged by CMake upstream, you should use `CMAKE_SIZEOF_VOID_P` instead. But that's no big problem.

Comment (o01eg) [reply CMakeLists.txt]: I could change it.

**Comment by o01eg** [comment]
I suppose for non XP toolchain should be set some minimal BOOST_USE_WINAPI_VERSION as well.

File: CMakeLists.txt (hunk 1 above)
Comment (o01eg) [reply CMakeLists.txt]: Fixed.

**Comment by adrianbroher** [comment]
LGTM

**Pull request merged**